DMCompiler/Compiler/DMPreprocessor/DMPreprocessorLexer.cs

```diff
@@ -182,17 +182,33 @@ class DMPreprocessorLexer : TextLexer {
                             case '*': {
                                 //Skip everything up to the "*/"
                                 Advance();
-                                while (true) {
-                                    bool isStar = GetCurrent() == '*';
-
-                                    if (isStar && Advance() == '/') {
+                                var comment_depth = 1;
+                                while (comment_depth > 0) {
+                                    if (GetCurrent() == '/') {
                                         Advance();
-                                        while (GetCurrent() == ' ' || GetCurrent() == '\t') {
+                                        if (GetCurrent() == '*') {
+                                            // We found another comment - up the nest count
+                                            comment_depth++;
                                             Advance();
+                                        } else if (GetCurrent() == '/') {
+                                            // Encountered a line comment - skip to end of line
+                                            while (Advance() != '\n' && !AtEndOfSource) ;
                                         }
-                                        break;
-                                    } else if (AtEndOfSource) return CreateToken(TokenType.Error, null, "Expected \"*/\" to end multiline comment");
-                                    else if (!isStar) Advance();
+                                    } else if (GetCurrent() == '*') {
+                                        if (Advance() == '/') {
+                                            // End of comment - decrease nest count
+                                            comment_depth--;
+                                            Advance();
+                                        }
+                                    } else if (AtEndOfSource) {
+                                        return CreateToken(TokenType.Error, null, "Expected \"*/\" to end multiline comment");
+                                    } else {
+                                        Advance();
+                                    }
+                                }
+
+                                while (GetCurrent() == ' ' || GetCurrent() == '\t') {
+                                    Advance();
                                 }
```

Comment:
Why skip whitespace after the comment?

`usr << (src./**/ x)`
I tested this line of code in BYOND, and it does not compile.

Reply:
Fixed - It was present before, was just more clear in the PR :)